src/capi.rs

```diff
@@ -444,6 +457,47 @@ pub unsafe extern fn mp4parse_get_track_video_info(parser: *mut mp4parse_parser,
     MP4PARSE_OK
 }
 
+// A fragmented file needs mvex table and contains no data in stts, stsc, and stco boxes.
+#[no_mangle]
+pub unsafe extern fn mp4parse_is_fragmented(parser: *mut mp4parse_parser, track_id: u32, fragmented: *mut u8) -> mp4parse_error {
+    if parser.is_null() || (*parser).poisoned() {
+        return MP4PARSE_ERROR_BADARG;
+    }
+
+    let context = (*parser).context_mut();
+    let tracks = &context.tracks;
+    (*fragmented) = false as u8;
+
+    if context.has_mvex == false {
+        return MP4PARSE_OK;
+    }
+
+    // check sample tables.
+    for track in tracks {
+        let fragmented_track = match track.track_id {
+            Some(id) => {
+                if id == track_id {
+                    track.fragmented
+                } else {
+                    (false, false, false)
```

Comment:
This replaces data from tracks we're not interested in with something expected to fail the subsequent match statement. Can you instead try pulling out the track of interest and testing that directly? Something like

```rust
match tracks.find(|track| track.track_id == track_id) {
    None => return MP4PARSE_ERROR_BADARG,
    Some(track) where track.fragmented == (true, true, true) => (*fragmented) = true,
    _ => (),
}
MP4PARSE_OK
```

You can split that into more lines if you like, the main issue I'd like to avoid is the implicit state passing. There's no reason to test tracks with different track_id's at all.